_posts/2019-07-16-neuron.md

```diff
@@ -0,0 +1,161 @@
+has been a surefire way to kill them. But in every field I have worked, the people who were best at the job were always
+the people who spent their free time continuing to pursue the hobby. So since I am not content with my mediocrity, 
+I have decided to try and breathe new life into this hobby. 
+
+# Foundation
+
+I really started digging into programming to do data analysis of the logs I kept while working at the climbing gym. 
+At the time, machine learning seemed like magic, a magic I very much wanted to possess. It was, however, _way_ beyond 
+my reach. But I've learned a thing or two (or four... or six) since then, and decided machine learning was a great 
+place to jump back in and see what I can do.
+
+The Python "Encyclopedia Galactica" of machine learning already exists. I'm not trying to replace it. I can't offer things
+cheaper than free, though I will inscribe the words "Don't Panic" on my machine learning library. ;)
+
+I am intentionally not using the big wigs of machine learning because my goal is not to write the worlds most accurate
+neural network. My goal is simply to understand the mechanics, and using a fully fleshed out library like Sci-kit Learn
+doesn't really afford me that opportunity. So here I go, reinventing the wheel. (My wheel is definitely still 
+kinda square... but we are working on it!) 
+
+## A kinda perceptive Perceptron
+
+My first step is writing just a single neuron. The goal of my standalone neuron, referred to as a Perceptron, is simple 
+classification. Take in any number of inputs (initially, we'll just use 2 inputs) and output -1 for Class A and 1 
+for Class B. The basic structure of a neuron is, on one end, all the receptors. These receptors feed their data to the
+"body" of the neuron, via weighted channels. In the body of the neuron, we calculate the weighted sum of all the data 
+from the receptors. This sum is passed into some kind of activation function that essentially handles translating 
+that sum into a usable value, and then spits that value out of our neuron.
+```
+input 1
+     \
+   weight 1
+       \
+        +-----------+
+        | weighted  |     +----------+
+        |           | --> |activation| --> classification value
+        |    sum    |     +----------+
+        +-----------+
+       /
+   weight 2
+     /
+input 2
+```
+### Step one: creating the neuron class
+This is some pretty straightforward code just setting up our skeleton on which to build.
+```python
+class Neuron:
+    def __init__(self, number_of_inputs):
+        self.number_of_inputs = number_of_inputs
+
+n = Neuron(number_of_inputs=2)
+```
+In order to do some simple error checking down the road, I chose not to make input numbers dynamic. This will let me do
+things like:
+```python
+if len(inputs) != self.number_of_inputs:
+    raise NeuronException('Length of inputs must be the same as the declared number of inputs.')
```

Comment:
Throughout, you may want to make these code highlighted sections only like ~40 characters wide:

![image](https://user-images.githubusercontent.com/2185159/61350196-a5c14180-a81b-11e9-9c80-6835e26d2156.png)

Reply:
40 characters wide is a huge restriction. Things start to look gross with all those '\'s for line continuations. I think its ok having to scroll in the code box on mobile to maintain code clarity.